### ISSUE TITLE
prevent broken builds in ec2_asg when using "replace all"

### ISSUE THREAD

**Issue by smiller171** [ansible/ansible#18929]
_From @smiller171 on April 2, 2015 15:4_

##### Issue Type:

Feature Idea
##### Ansible Version:

1.9.0.1
##### Environment:

Ubuntu server talking to AWS API
##### Summary:

Currently the "replace all" method in ec2_asg allows you to publish broken builds to an auto-scaling group.
##### Steps To Reproduce:

Create an auto-scaling group with a web server that listens on port 80, with a health check watching port 80.

Update that auto-scaling group with an AMI that has port 80 blocked.
##### Expected Results:

Health check will fail, ec2_asg module terminates failed instance, restores previous Launch configuration to ASG, task is marked as failed.
##### Actual Results:

All good instances are replaced with bad instances, creating downtime. Task is marked as successful.

_Copied from original issue: ansible/ansible-modules-core#1066_

**Comment by smiller171** [comment]
The current behavior is a problem, because the auto-scaling group will continue to replace the bad instance(s) making it impossible to fix the situation by running the task again with a good AMI. Attempting to do so results in an error:

"The age of the instance (X seconds) is less than the configured grace period of Y seconds"

**Comment by smiller171** [comment]
_From @mscherer on April 16, 2015 16:32_

Could you please change the title of this feature request to be more descriptive than "feature request" ?

**Comment by smiller171** [comment]
@bcoca I think it may be more descriptive to say "enable failure detection and rollback when using "replace all" "

**Comment by smiller171** [comment]
@mscherer @bcoca Is there any movement on this? I've been forced to switch to the CloudFormation module to avoid this issue.

**Comment by smiller171** [comment]
_From @ansibot on August 1, 2016 3:30_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by smiller171** [comment]
@bcoca @defionscode This issue has been open for over a year and it's possible that things have changed and it's no longer an issue. I can take the time to test if needed

**Comment by smiller171** [comment]
_From @ansibot on September 13, 2016 1:39_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by smiller171** [comment]
_From @ansibot on October 6, 2016 16:8_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by smiller171** [comment]
_From @ansibot on December 7, 2016 18:1_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
@smiller171 Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibotdev** [comment]
@smiller171 You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
